spec/models/access_spec.rb

```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe Access, type: :model do
+  describe "Validations" do
+    it { is_expected.to validate_presence_of(:mode) }
+  end
+
+  describe "Associations" do
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:scope).optional }
+  end
```

Comment:
Let's add the missing specs for allowed scope types (and any additional validations based on PR feedback) here as well.